apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -72,7 +72,11 @@ export function useSetupTradeState(): void {
         }
       }
 
-      tradeNavigate(providerChainId, getDefaultTradeRawState(providerChainId))
+      // Adding timeout of 500ms to avoid running into infinity loop of updating provider and url state.
+      // issue GH : https://github.com/cowprotocol/cowswap/issues/4734
+      setTimeout(() => {
+        tradeNavigate(providerChainId, getDefaultTradeRawState(providerChainId))
+      }, 500)
```

Comment:
Should the timeout be cleared when the useEffect is ummounted?

Reply:
Yup you are right!, I've now moved to debounce approach to prevent subsequent function calls with @anxolin suggestion.